xls/dslx/stdlib/apfloat.x

```diff
@@ -183,6 +183,183 @@ pub fn unflatten<EXP_SZ:u32, SFD_SZ:u32,
   }
 }
 
+pub fn cast_from_fixed<EXP_SZ:u32, SFD_SZ:u32, UEXP_SZ:u32 = EXP_SZ + u32:1,
+  NUM_SRC_BITS:u32, EXTENEDED_SFD_SZ:u32 = SFD_SZ + NUM_SRC_BITS>
+  (to_cast: sN[NUM_SRC_BITS]) -> APFloat<EXP_SZ, SFD_SZ> {
+  // Determine sign.
+  let sign = to_cast[NUM_SRC_BITS-u32:1 : NUM_SRC_BITS];
+
+  // Determine exponent.
+  let abs_magnitude = (to_cast if sign == u1:0 else -to_cast) as uN[NUM_SRC_BITS];
+  let lz = clz(abs_magnitude);
+  let num_trailing_nonzeros = NUM_SRC_BITS as uN[NUM_SRC_BITS] - lz;
+  let exp = num_trailing_nonzeros as uN[UEXP_SZ] - uN[UEXP_SZ]:1;
```

Comment:
Can you parenthesize this? I (and I assume others?) can never keep precedence rules in my head.

Reply:
And it looks like you use "uN[UEXP_SZ] - uN[UEXP_SZ]:1" > 1x; consider making a temp to hold it.

Reply:
Parentheses parenthesized.
For "uN[UEXP_SZ] - uN[UEXP_SZ]:1" , we are casting the value on the left hand side and then subtracting 1.  I don't believe there are common terms here (we cast num_trailing_nonzeros and subtract 1 on line 196 and cast EXP_SZ and subtract 1 on line 197).